UnityProject/Assets/Scripts/Items/Weapons/Explosive.cs

```diff
@@ -0,0 +1,190 @@
+		[Header("Explosive Components")]
+		[SerializeField] private SpriteHandler spriteHandler;
+		[SerializeField] private SpriteDataSO activeSpriteSO;
+		[SerializeField] private RegisterItem registerItem;
+		[SerializeField] private ObjectBehaviour objectBehaviour;
+		[SerializeField] private Pickupable pickupable;
+		[SerializeField] private HasNetworkTabItem explosiveGUI;
+		[HideInInspector] public GUI_Explosive GUI;
+
+		private bool hasExploded;
+		private bool isArmed;
+		private bool countDownActive = false;
+		private bool isOnObject = false;
+
+		public int TimeToDetonate
+		{
+			get => timeToDetonate;
+			set => timeToDetonate = value;
+		}
+
+		public bool IsArmed
+		{
+			get => isArmed;
+			set => isArmed = value;
+		}
+
+		public int MinimumTimeToDetonate => minimumTimeToDetonate;
+		public bool DetonateImmediatelyOnSignal => detonateImmediatelyOnSignal;
+		public bool CountDownActive => countDownActive;
+		public ExplosiveType ExplosiveType => explosiveType;
+
+		public async void Countdown()
+		{
+			countDownActive = true;
+			spriteHandler.SetSpriteSO(activeSpriteSO);
+			if (GUI != null) GUI.StartCoroutine(GUI.UpdateTimer());
+			await Task.Delay(timeToDetonate * 1000); //Delay is in milliseconds
+			Detonate();
+		}
+
+		private void Detonate()
+		{
+			//We don't use Explosion.StartExplosion() because it doesn't look or work as
+			//Explosion prefabs do
+			if (hasExploded)
+			{
+				return;
+			}
+			hasExploded = true;
+
+			if (isServer)
+			{
+				// Get data before despawning
+				var explosionMatrix = registerItem.Matrix;
+				var worldPos = objectBehaviour.AssumedWorldPositionServer();
+
+				// Despawn the explosive
+				_ = Despawn.ServerSingle(gameObject);
+
+				// Explosion here
+				var explosionGO = Instantiate(explosionPrefab, explosionMatrix.transform);
+				explosionGO.transform.position = worldPos;
+				explosionGO.Explode(explosionMatrix);
+			}
+		}
+
+		/// <summary>
+		/// Toggle the detention mode of the explosive, true means it will only work with a signaling device.
+		/// </summary>
+		/// <param name="mode"></param>
+		public void ToggleMode(bool mode)
+		{
+			detonateImmediatelyOnSignal = mode;
+		}
+
+		public override void ReceiveSignal(SignalStrength strength)
+		{
+			if(countDownActive == true || isArmed == false) return;
+			if (detonateImmediatelyOnSignal)
+			{
+				Detonate();
+				return;
+			}
+			Countdown();
+		}
+
+		/// <summary>
+		/// checks to see if we can attach the explosive to an object.
+		/// </summary>
+		private bool CanAttatchToTarget(Matrix matrix, RegisterTile tile)
+		{
+			return matrix.Get<RegisterDoor>(tile.WorldPositionServer, true).Any() || matrix.Get<Pickupable>(tile.WorldPositionServer, true).Any();
+		}
+
+		#region Interaction
+
+		public bool WillInteract(MouseDrop interaction, NetworkSide side)
+		{
+			if (interaction.TargetObject.TryGetComponent<SignalEmitter>(out var _) && interaction.IsFromInventory) return true;
```

Comment:
this WillInteract is missing the Default validation

Reply:
no no, it's meant to be like that because `Default` doesn't work with inventories, so i made a work around and did `interaction.IsFromInventory`